### PR TITLE
Allow adding custom container apps

### PR DESCRIPTION
**Pull request by Stretch96**
* This will allow adding Sidecar like containers within the container app environment
* They can optionally be launched in a different container app environment too

### CODE REVIEW

**Review by ryantk — APPROVED** [review]

**Comment by ryantk** [comment]
I am testing this currently, but having a few unrelated issues with my setup so its taking a while

**Comment by DrizzlyOwl** [comment]
@Stretch96 can you please add usage instructions to the readme?

**Comment by Stretch96** [comment]
Added in some docs in the Readme 👍 @DrizzlyOwl

**Review by DrizzlyOwl — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -675,3 +675,29 @@ variable "container_app_blob_storage_ipv4_allow_list" {
...
+          activeRevisionsMode = optional(string, "single")
+          secrets             = optional(list(map(string)), [])
+          ingress             = optional(any, {})
+          registries          = optional(list(map(any)), [])
+          dapr                = optional(map(string), {})
+        })
+        template = object({
+          revisionSuffix = string
+          containers     = list(any)
+          scale          = map(number)
```
Comment (Stretch96) [inline variables.tf:696]: we should set this as `any`, because it can contain strings or lists

**Pull request merged**